Compass/CompassViewController.swift

```diff
@@ -10,59 +10,86 @@ import UIKit
 import CoreLocation
 import Parse
 
-class ViewController: UIViewController, CLLocationManagerDelegate {
+class CompassViewController: UIViewController, CLLocationManagerDelegate {
 
     var locationManager = CLLocationManager()
-    var currentLocation: CLLocation? = nil
-    var targetLocation: CLLocation? = nil
-    let par = PServer() 
+    var currentLocation: CLLocation?
+    var targetLocation: CLLocation?
+    let par = PServer()
     let myCalc = Calculations()
     var myUserData = UserData()
     
     let ArrowImage = UIImageView(image: UIImage(named: "240px-Green_Arrow_Up_Darker.png"))
     let degreeLabel = UILabel(frame: CGRect.zero)
+    let distanceLabel = UILabel(frame: CGRect.zero)
     let setTargetLocationButton = UIButton(frame: CGRect.zero)
+    let backButton = UIButton(frame: CGRect.zero)
     
     
     func initLocationManager() {
-        
-        //this function must be moved somewhere else, call once when app started
-        //par.initParse(appID: "492795c6ea25112881915677092fb19d95f43ce0", clKey: "6c4448eb0dc5d344a0ca35f8d8f978ff82b76028", serverAddress: "http://18.188.82.67:80/parse")
-        
-        if let tempID = UserDefaults.standard.string(forKey: "parseObjectID") {
-            myUserData.objectID = tempID
-        } else { myUserData.objectID = "" }
+        //Fault???
         if let userName = UserDefaults.standard.string(forKey: "userName") {
             myUserData.name = userName
-        } else { myUserData.objectID = "" }
-        
-        
+        } else { myUserData.name = ""
+        }
+        if let objectID = UserDefaults.standard.string(forKey: "locationObjectId") {
+            myUserData.objectID = objectID
+        } else { myUserData.objectID = ""
+        }
+    
         locationManager.delegate = self
         locationManager.desiredAccuracy = kCLLocationAccuracyBest
         locationManager.requestWhenInUseAuthorization()
         locationManager.startUpdatingLocation()
         locationManager.startUpdatingHeading()
     }
     
-    
     override func viewDidLoad() {
         super.viewDidLoad()
         initLocationManager()
         createUI()
+        setUpHandlers()
+        
     }
     
-    func createUI(){
+    override func viewDidAppear(_ animated: Bool) {
```

Comment:
This function is great! Great use of `if let` and great formatting.
I might be inclined to change one of the if lets to a guard though